src/basic/fs-util.c

```diff
@@ -374,6 +374,10 @@ int fd_warn_permissions(const char *path, int fd) {
 int touch_fd(int fd, usec_t stamp) {
         assert(fd >= 0);
 
+        /* Return early if there's no timestamp to update */
+        if (stamp == 0)
+                return 0;
```

Comment:
(just to mention this: this is a bit too simple, because in many situations people actually do set the timestamp to the unix epoch (which 0 means), hence we shouldn't use this as "niche" value for such a low-level function here...